frontend/src/test/java/com/google/tripmeout/frontend/servlet/PlaceParentServletTest.java

```diff
@@ -0,0 +1,222 @@
+package com.google.tripmeout.frontend.servlet;
+
+import static com.google.common.truth.Truth.assertThat;
+import static com.google.common.truth.Truth8.assertThat;
+import static org.mockito.Mockito.when;
+import static org.mockito.MockitoAnnotations.initMocks;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.google.maps.GeoApiContext;
+import com.google.maps.PlaceDetailsRequest;
+import com.google.maps.errors.ApiException;
+import com.google.maps.model.PlaceDetails;
+import com.google.tripmeout.frontend.PlaceVisitModel;
+import com.google.tripmeout.frontend.error.TripMeOutException;
+import com.google.tripmeout.frontend.serialization.GsonModelSerializationModule;
+import com.google.tripmeout.frontend.servlets.PlaceParentServlet;
+import com.google.tripmeout.frontend.storage.InMemoryPlaceVisitStorage;
+import com.google.tripmeout.frontend.storage.PlaceVisitStorage;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.StringReader;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+
+public class PlaceParentServletTest {
+  @Mock HttpServletRequest request;
+  @Mock PlaceDetailsRequest detailsRequest;
+  private Gson gson;
+
+  private final GeoApiContext CONTEXT =
+      new GeoApiContext.Builder().apiKey("AIzaSyBbXCXC2uWv3baNmirLtqUYbFsFwCXqLV8").build();
```

Comment:
Tests should never contain API keys like this. Luckily, the need for inline contexts will go away once we add a place service interface anyway.